### ISSUE TITLE
Generator/Discriminator Combined Loss Values Question

### ISSUE THREAD

**Issue by coded5282** [junyanz/pytorch-CycleGAN-and-pix2pix#1028]
Hi, thanks a lot for the code. It's very helpful! I have a question regarding the CycleGAN loss functions. Why is it that the code combines the loss values for both generators but then calculates the loss values for the discriminators separately?

Is there any downside/upside to combining the loss values for the 2 generator models? I've seen some other implementations where they split the loss value into 2 - one for generator A and the other for generator B. It seems to me that splitting the loss value for each generator might make it easier for the gradients to flow, but that's just a guess. Happy to hear your thoughts!

**Comment by junyanz** [comment]
It should be the same. The D is calculated separately as we can implement a common function `backward_D_basic` and call it twice.

**Comment by coded5282** [comment]
@junyanz Thank you for your quick response!
